packages/react-router-dom/__tests__/data-browser-router-test.tsx

```diff
@@ -5712,10 +5712,10 @@ function testDomRouter(
             await waitFor(() => screen.getByText("Page (1)"));
             expect(getHtml(container)).toMatch("Num fetchers: 1");
 
-            // Resolve after the navigation and revalidation
+            // Resolve action after the navigation and trigger revalidation
             dfd.resolve("FETCH");
             await waitFor(() => screen.getByText("Num fetchers: 0"));
-            expect(getHtml(container)).toMatch("Page (1)");
+            expect(getHtml(container)).toMatch("Page (2)");
```

Comment:
This test was incorrect before due to the bug.  The revalidation would cause the count to increment again after the navigation completed